libs/dotcms-webcomponents/src/components/contenttypes-fields/dot-key-value/components/key-value-table/key-value-table.tsx

```diff
@@ -38,6 +44,124 @@ export class KeyValueTableComponent {
         );
     }
 
+    componentDidLoad() {
+        this.bindDraggableEvents();
+    }
+
+    componentDidUpdate() {
+        this.bindDraggableEvents();
+    }
+
+    // D&D - BEGIN
+
+    private bindDraggableEvents() {
+        const rows = document.querySelectorAll('key-value-table tr');
+        rows.forEach((row) => {
+            row.setAttribute('draggable', 'true');
+
+            row.removeEventListener('dragstart', this.handleDragStart.bind(this), false);
+            row.removeEventListener('dragenter', this.handleDragEnter, false);
+            row.removeEventListener('dragover', this.handleDragOver.bind(this), false);
+            row.removeEventListener('dragleave', this.handleDragLeave, false);
+            row.removeEventListener('drop', this.handleDrop.bind(this), false);
+            row.removeEventListener('dragend', this.handleDragEnd.bind(this), false);
+
+            row.addEventListener('dragstart', this.handleDragStart.bind(this), false);
+            row.addEventListener('dragenter', this.handleDragEnter, false);
+            row.addEventListener('dragover', this.handleDragOver.bind(this), false);
+            row.addEventListener('dragleave', this.handleDragLeave, false);
+            row.addEventListener('drop', this.handleDrop.bind(this), false);
+            row.addEventListener('dragend', this.handleDragEnd.bind(this), false);
+        });
```

Comment:
Why not using event delegation here? you can just listen to the `table`. https://www.mattlunn.me.uk/2012/05/event-delegation-in-javascript/

Reply:
since I need to detect the position while the row is being dragged to put the placeholder(position) it would be even more complicated to attach the event to the whole container (table)